dds_web/utils.py

```diff
@@ -571,3 +572,49 @@ def calculate_version_period_usage(version):
         version.time_invoiced = now
 
     return bytehours
+
+
+# maintenance check
+def block_if_maintenance():
+    """Block API requests if maintenance is ongoing and projects are busy."""
+    # Get maintenance row
+    maintenance: models.Maintenance = models.Maintenance.query.first()
+
+    # Possibly block request if maintenance ongoing / planned
+    if maintenance.active:
+        # Endpoints accepting requests during active maintenance / planned
+        project_required_endpoints: typing.List = [
+            f"/api/v1{resource}" for resource in ["/file/new", "/file/update", "/proj/busy"]
+        ]
+        admin_endpoints: typing.List = [
+            f"/api/v1{x}"
+            for x in [
+                "/user/info",
+                "/maintenance",
+                "/unit/info/all",
+                "/motd",
+                "/motd/send",
+                "/proj/busy/any",
+            ]
+        ]
+        approved_endpoints: typing.List = project_required_endpoints + admin_endpoints
+
+        # Request not to accepted endpoint
+        # OR request to accepted endpoint but project not specified or busy
+        current_endpoint: str = flask.request.path
+        if current_endpoint not in approved_endpoints:
+            # Request not accepted during maintenance
+            raise MaintenanceOngoingException()
+        else:
+            # Request accepted during maintenance but...
+            req_args = flask.request.args
+            if current_endpoint in project_required_endpoints:
+                # Request requires a project
+                if not (req_args and (project_id := req_args.get("project"))):
+                    raise MaintenanceOngoingException()
+
+                # Request requires an busy project
```

Comment:
# Request requires an busy project
should be
# Request requires a busy project